File: spring-backend/src/main/java/com/credigo/backend/dto/UserResponse.java
package com.credigo.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserResponse {

  private Integer id; // The user's ID from the database
  private String username;
  private String email;
  private String phoneNumber;
  private LocalDate dateOfBirth;
  private LocalDateTime createdAt;
  // Notice: NO password field here!

  private java.util.Set<String> roles;

  public java.util.Set<String> getRoles() {
    return roles;
  }

  public void setRoles(java.util.Set<String> roles) {
    this.roles = roles;
  }
}

File: mobile/credigo/app/src/main/java/com/sia/credigo/model/UserResponse.kt
package com.sia.credigo.model

import java.math.BigDecimal

data class UserResponse(
    val id: Int,
    val username: String,
    val email: String,
    val phoneNumber: String?,
    val dateOfBirth: String?,
    val createdAt: String,
    val roles: Set<String>,
    val active: Boolean?,
    val balance: BigDecimal?
)
